src/pages/api/usecases/send-message-usecase.js

```diff
@@ -0,0 +1,54 @@
+import {  GovNotifyGateway } from "../../../gateways/gov-notify-api-gateway";
+import { TEST_AND_TRACE_FOLLOWUP_EMAIL, TEST_AND_TRACE_FOLLOWUP_TEXT, PRE_CALL_MESSAGE_TEMPLATE } from "../../../helpers/constants";
+export class SendMessageUseCase{
+  async sendMessage( pathSegments, queryParams){
+    const govNotifyGateway = new GovNotifyGateway()
+      if(queryParams.phoneNumber){
+        try {
+          const templateId = (pathSegments[0] == TEST_AND_TRACE_FOLLOWUP_TEXT) ? process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_TEXT : ""
+          let response = await govNotifyGateway.sendSms(templateId, queryParams.phoneNumber, {})
+          return response
+        } catch (error) {
+          console.log(`Send text usecase error: ${error}`)
+          return error
+        }
+      }
+      if(queryParams.email){
+        try {
+          const templateId = (pathSegments[0] == TEST_AND_TRACE_FOLLOWUP_EMAIL) ? process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_EMAIL : ""
+          let response = await govNotifyGateway.sendEmail(templateId, queryParams.email)
+          return response;
+        } catch (error) {
+          console.log(`Send email usecase error: ${error}`)
+          return error
+        }
+      }
+
+      if(pathSegments[0] == "previewTemplate"){
+        try {
+          if((queryParams.templateType == TEST_AND_TRACE_FOLLOWUP_TEXT)){
+
+            let templateId =  process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_TEXT
```

Comment:
Is it possible to access the template id by queryParams.templateType instead of having 3 if blocks? 
Not sure about the syntax, but something like this:
```let templateId =  process.env.[queryParams.templateType]```
and then you only need to check if the ```queryParams.templateType``` exists

Reply:
I dont think so, i had a go at this and i think the only way to access env vars is `process.env.SOME_ENV_VAR`

Reply:
Fair.
Maybe having switch block instead of if blocks would be more readable?
Also, templateId could be declared outside of the if/switch block and this could be the only line within the conditional statement 
```
templateId =  process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_TEXT
```

So something like this:
```
try {
            let templateId;
            switch(queryParams.templateType) {
              case TEST_AND_TRACE_FOLLOWUP_TEXT:
                templateId =  process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_TEXT
                break;
              case TEST_AND_TRACE_FOLLOWUP_EMAIL:
                templateId = process.env.TEST_AND_TRACE_FOLLOWUP_TEMPLATE_EMAIL
                break;
               case PRE_CALL_MESSAGE_TEMPLATE:
                templateId = process.env. PRE_CALL_MESSAGE_TEMPLATE
                break;
            }
             const response = await govNotifyGateway.getTemplatePreview(templateId, {name:"(first name)"})
             return response;
         } catch (error) {
           console.log(`Get template usecase error: ${error}`)
           return error
         }
         ```

Reply:
Or the same with If statements

Reply:
Essentially this could be extracted outside of if blocks cause it's being executed in every case:
```
        const response = await govNotifyGateway.getTemplatePreview(templateId, {name:"(first name)"})
             return response;
```

Reply:
Oh sorry! You already did that!